### PR TITLE
VideoStreaming does not start during HMI Level "LIMITED"

### LINKED ISSUE #1289: VideoStreaming does not start during HMI Level "LIMITED".
### Bug Report
VideoStreaming does not start during HMI Level "LIMITED". 
In version 4.7.2, VideoStreaming can start only when the HMI Level is "FULL" by the check logic in Proxy.
However, an error is returned when the HMI Level is "LIMITED". As a result, video streaming cannot be started.

##### Reproduction Steps
1. Start a video application (HMI Level "FULL" is notified at this time)
2.Then,  start a media application (HMI Level "LIMITED" is notified to the video application at this time)
3.At the timing of step two, start video streaming.

##### Expected Behavior
VideoStreaming should start even if HMI Level = "LIMITED".

##### Observed Behavior
Error is returned, therefore, Video Streaming cannot be started.

##### OS & Version Information
* SDL Android Version: 4.7.2

### PR DESCRIPTION
**Pull request by JulianKast**
Fixes #1289 

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR (or explain why below)
- [x] I have run the unit tests with this PR
- [x] I have tested this PR against Core and verified behavior (if applicable, if not applicable, explain why below).
- [x] I have tested Android, Java SE, and Java EE

#### Core Tests
Tested using sync version: 3.0 19205_DEVTEST

### Summary
Added checks in VideoStreamManager to allow video streaming to start in hmiLevel Limited.

To Test in JsonRPCMarshaller.java I edited Method "unmarshall(byte[] message)" JSON to set HMI Level to Limited if it is full

```
			if(jsonObject.has("hmiLevel")){
				if(jsonObject.get("hmiLevel").equals("FULL")){
					jsonObject.remove("hmiLevel");
					jsonObject.put("hmiLevel", "LIMITED");
				}
			}
```

##### Bug Fixes
* Aligns with IOS and allows video streaming to start in hmi limited

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java [hunk 1]
```diff
@@ -303,7 +303,7 @@ public void onError(String info) {
 	 */
 	protected void startStreaming(VideoStreamingParameters parameters, boolean encrypted){
 		this.parameters = parameters;
-		if(hmiLevel != HMILevel.HMI_FULL){
+		if(hmiLevel != HMILevel.HMI_FULL && hmiLevel != HMILevel.HMI_LIMITED){
 			Log.e(TAG, "Cannot start video service if HMILevel is not FULL.");
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: The error message is not accurate anymore cause video can stream in limited

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java [hunk 2]
```diff
@@ -205,7 +205,7 @@ private synchronized void checkState(){
 		if(this.getState() == SETTING_UP
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: Shouldn't we also check alow calling `checkState()` method on line 154 when level = LIMTED?

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java [hunk 3]
```diff
@@ -205,7 +205,7 @@ private synchronized void checkState(){
...
 				&& isTransportAvailable
 				&& hmiLevel != null
-				&& hmiLevel.equals(HMILevel.HMI_FULL)
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: Do `isStreaming()` and is `isPaused()` need to be updated as well? Please double-check how iOS handle that. Also, all usages for HMILevel in this class should be revisited to make sure they make sense for streaming video in `LIMITED`.

**Comment by bilal-alsharifi** [comment]
I also suggest making two lists: `streamableLevels = {FULL, LIMITED}` and `noneStreamableLevels = {NONE, BACKGROUND}` and use `list.contains()` method to check if the current level supports streaming or not instead of comparing with each state individually each time. That will make the code easier to read and less error prone.

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java (hunk 2 above)
Comment (JulianKast) [reply android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: Agreed! Added

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java (hunk 3 above)
Comment (JulianKast) [reply android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: Agreed! Added!

**Comment by JulianKast** [comment]
@bilal-alsharifi I like the Idea of the two list for hmi Levels, I added them.

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java [hunk 4]
```diff
@@ -95,6 +97,8 @@
 	private IVideoStreamListener streamListener;
 	private boolean isTransportAvailable = false;
 	private boolean hasStarted;
+	private List<HMILevel> streamableLevels = new ArrayList<>(Arrays.asList(HMILevel.HMI_FULL, HMILevel.HMI_LIMITED));
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: Can we simplify it like that? also applies to the other list
```suggestion
	private List<HMILevel> streamableLevels = Arrays.asList(HMILevel.HMI_FULL, HMILevel.HMI_LIMITED);
```

File: android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java [hunk 5]
```diff
@@ -73,7 +73,9 @@
 import com.smartdevicelink.util.Version;
 
 import java.lang.ref.WeakReference;
+import java.lang.reflect.Array;
```
Comment (bilal-alsharifi) [inline android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java]: this import is unused

**Pull request merged**